hhs_facilities/delphi_hhs_facilities/geo.py

```diff
@@ -38,3 +39,39 @@ def convert_geo(df: pd.DataFrame, geo: str, gmpr: GeoMapper) -> pd.DataFrame:
         output_df = gmpr.add_geocode(df, "fips", geo, from_col="fips_code")
         output_df["geo_id"] = output_df[geo]
     return output_df
+
+
+def fill_missing_fips(df: pd.DataFrame, gmpr: GeoMapper) -> pd.DataFrame:
+    """
+    Fill in missing FIPS code if zip is present.
+
+    Maps rows that have the FIPS missing but zip present. The rest of the rows,
+    including those where both FIPS and zip are nan, are kept as is and appended back at the end.
+
+    TODO #636 Generalize this function to geomapper.
+
+    Parameters
+    ----------
+    df: pd.DataFrame
+        Input DataFrame containing zip and fips columns.
+    gmpr:
+        GeoMapper object.
+
+    Returns
+    -------
+    DataFrame with missing FIPS imputed with zip.
+    """
+    mask = pd.isna(df["fips_code"]) & ~pd.isna(df["zip"])
+    no_fips = df[mask]
+    fips_present = df[~mask]
+    no_data_cols = [c for c in df.columns if df[c].dtypes not in (dtype("int64"), dtype("float64"))]
+    data_cols = list(set(df.columns) - set(no_data_cols))
+    # we keep unmapped zips because they may be used in HRR mapping so we don't want to drop them
```

Comment:
> we keep unmapped zips because they may be used in HRR mapping

is that actually true? how many zips have HHRs but not FIPS codes?

Reply:
I saw one zip that didn't get mapped to a fips: 00935 (in puerto rico). It's also not in our HRR mapping file, but I figure it _could_ be there so it would be better to keep, and otherwise it would just get dropped during HRR mapping.

The other thing this gives us is consistent column sums between a df before and after imputation, so this function is just responsible for filling in values where possible and ignoring otherwise. Then any dropping of values is handled by the later geomapping. I can update the comment accordingly